src/Data/Bifunctor/Biap.hs

```diff
@@ -0,0 +1,128 @@
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DeriveTraversable          #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+
+#include "bifunctors-common.h"
+#ifndef MIN_VERSION_semigroups
+#define MIN_VERSION_semigroups(x,y,z) 0
+#endif
+
+-----------------------------------------------------------------------------
+-- |
+-- Copyright   :  (C) 2008-2016 Edward Kmett
+-- License     :  BSD-style (see the file LICENSE)
+--
+-- Maintainer  :  Edward Kmett <ekmett@gmail.com>
+-- Stability   :  provisional
+-- Portability :  portable
+--
+----------------------------------------------------------------------------
+module Data.Bifunctor.Biap
+ ( Biap(..)
+ ) where
+
+import GHC.Generics
+import qualified Control.Monad as M
+import qualified Control.Applicative as A
+#if __GLASGOW_HASKELL__ < 710
+import Data.Foldable
+import Data.Monoid
+import Data.Traversable
+#endif
+import Data.Biapplicative
+import Data.Bifoldable
+#if LIFTED_FUNCTOR_CLASSES
+import Data.Functor.Classes
+#endif
+#if MIN_VERSION_base(4,9,0) || MIN_VERSION_semigroups(0,16,2)
+import qualified Data.Semigroup as S
+#endif
+#if __GLASGOW_HASKELL__ >= 800
+import qualified Control.Monad.Fail as Fail (MonadFail)
+#endif
+import GHC.Base hiding (Any)
```

Comment:
This is a strange import. I _think_ it's being used to import `Alternative`, so I think importing `Control.Applicative` would be a more direct way of accomplishing this.